androidApp/src/main/java/com/mbta/tid/mbta_app/android/search/SearchBarOverlay.kt

```diff
@@ -0,0 +1,146 @@
+            currentNavEntry?.arguments?.getString("stopId")?.isBlank() ?: true
+        }
+    var expanded by rememberSaveable { mutableStateOf(false) }
+    var searchInputState by rememberSaveable { mutableStateOf("") }
+    val searchResults = getSearchResults(searchInputState)
+
+    fun handleSearch(stopId: String) {
+        expanded = false
+        searchInputState = ""
+        onStopNavigation(stopId)
+    }
+
+    val buttonColors =
+        ButtonColors(
+            containerColor = colorResource(R.color.fill3),
+            disabledContainerColor = colorResource(R.color.fill3),
+            contentColor = colorResource(R.color.deemphasized),
+            disabledContentColor = colorResource(R.color.deemphasized),
+        )
+
+    Box(contentAlignment = Alignment.TopCenter) {
+        Box(
+            modifier =
+                Modifier.absoluteOffset {
+                        if (expanded) IntOffset(0, 0) else IntOffset(0, 12.dp.roundToPx())
+                    }
+                    .zIndex(1f),
+            contentAlignment = Alignment.Center,
+        ) {
+            if (visible) {
+                Box(
+                    modifier =
+                        Modifier.absoluteOffset(y = 3.5.dp)
+                            .height(64.dp)
+                            .width(364.dp)
+                            .border(2.dp, colorResource(R.color.halo), RoundedCornerShape(10.dp))
+                )
+                SearchBar(
+                    shape = RoundedCornerShape(10.dp),
+                    colors =
+                        SearchBarDefaults.colors(containerColor = colorResource(R.color.fill3)),
+                    inputField = {
+                        SearchBarDefaults.InputField(
+                            colors =
+                                SearchBarDefaults.inputFieldColors(
+                                    focusedTextColor = colorResource(R.color.deemphasized),
+                                    unfocusedTextColor = colorResource(R.color.deemphasized),
+                                    focusedPlaceholderColor = colorResource(R.color.deemphasized),
+                                    unfocusedPlaceholderColor = colorResource(R.color.deemphasized),
+                                ),
+                            query = searchInputState,
+                            placeholder = { Text(stringResource(R.string.search_by_stop)) },
+                            expanded = expanded,
+                            onQueryChange = { searchInputState = it },
+                            onExpandedChange = { expanded = it },
+                            onSearch = {
+                                expanded = false
+                                if (
+                                    searchResults?.stops != null && searchResults.stops.isNotEmpty()
+                                ) {
+                                    handleSearch(searchResults.stops[0].id)
```

Comment:
question: selecting the first result by default is unexpected to me - is that an android-specific pattern?

Reply:
on the iOS-side, hitting the "done" button on search does nothing (or maybe re-runs the search in the background? but the effect is the same - you still see a list of search results to choose from, rather than automatically being taken to the first result)